### PR TITLE
Chris news 6 - adding a comments list 

### PR DESCRIPTION
**Pull request by ChrisMaylor92**

### CODE REVIEW

File: src/components/ArticleComments.jsx [hunk 1]
```diff
@@ -1,3 +1,49 @@
...
+
+    if (loading) {
+        return <div>Loading!</div>;
+    }
+
+
+    return <div>
+        <h2>{article.title}</h2>
+        <h2>Comments</h2>
+        <PostComment setComments={setComments} article_id={article_id}/>
```
Comment (2202Hannah) [inline src/components/ArticleComments.jsx]: There is a post comment component here but there is currently no comment posting functionality on your app?

File: src/components/ArticleComments.jsx [hunk 2]
```diff
@@ -1,3 +1,49 @@
...
-    return <h1>Article Comments!</h1>
+
+    const [comments, setComments] = useState([])
+    const [article, setArticle] = useState({})
+    const [loading, setLoading] = useState(true);
+    
+    const { article_id } = useParams();
+
+    useEffect(() => {
+        getCommentsByArticleId(article_id)
```
Comment (2202Hannah) [inline src/components/ArticleComments.jsx]: this is nicely done :)

File: src/components/ArticleComments.jsx [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+import { getArticleById, getCommentsByArticleId } from "../API"
+import { useParams, Link } from "react-router-dom";
+import { CommentCard } from "./CommentCard";
+import { PostComment } from "./PostComment";
+
 export const ArticleComments = () => {
-    return <h1>Article Comments!</h1>
+
+    const [comments, setComments] = useState([])
+    const [article, setArticle] = useState({})
```
Comment (2202Hannah) [inline src/components/ArticleComments.jsx]: If ArticleComments is a child of the Article component you could pass article down as a prop so wouldn't need to set the state again here

**Review by 2202Hannah — CHANGES_REQUESTED** [review]
Hi Chris, a solid start to this ticket! 
- 👍 User is given visual feedback if data is still loading
- We currently have a link to a new route in order to view the article comments on a different page. However this task asks for the list of comments for each article to be on the article page. Perhaps you could render the comments component underneath the article instead?
- When an article has no comments it would be nice to provide the user with some feedback (eg. article 37)
- You have implemented comment voting which is cool but not actually a requirement for this ticket. If you are going to include voting in this merge, I would suggest that the votes should be handled with optimistic rendering if possible and you can find an example of that in the notes here https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering. However I would suggest leaving this functionality to one side for now and not including it in this merge as comment voting is not part of the core tasks.

File: src/components/CommentCard.jsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    
+    return <div>
+        <p>{comment.body}</p>
+        <p>Author: {comment.author}</p>
+        <p>Votes: {voteCount}</p>
+        <button onClick={handleVote}>{hasVoted? "Remove" : "Add"} Vote</button>
+        {err ? <p>{err}</p> : null}
+    </div>
+}
```
Comment (itskatherine) [inline src/components/CommentCard.jsx:42]: Hullooo, 
- you could set the err back to null when the vote change has been successful - so if there's a problem with the internet and then it's fixed you're not left with the error message
- Nice implementation of optimistic rendering here

**Review by itskatherine — COMMENTED** [review]
Hi Chris, nice work have a look over my comment then you're good to merge

**Pull request merged**